Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingAccelerationStructurePass.cpp

```diff
@@ -41,20 +42,104 @@ namespace AZ
+
+                    uint32_t instanceIndex = 0;
+                    for (auto& subMesh : subMeshes)
+                    {
+                        tlasDescriptorBuild->Instance()
+                            ->InstanceID(instanceIndex)
+                            ->InstanceMask(subMesh.m_mesh->m_instanceMask)
+                            ->HitGroupIndex(0)
+                            ->Blas(subMesh.m_blas)
+                            ->Transform(subMesh.m_mesh->m_transform)
+                            ->NonUniformScale(subMesh.m_mesh->m_nonUniformScale)
+                            ->Transparent(subMesh.m_material.m_irradianceColor.GetA() < 1.0f);
+
+                        instanceIndex++;
+                    }
+
+                    unsigned proceduralHitGroupIndex = 1; // Hit group 0 is used for normal meshes
+                    const auto& proceduralGeometryTypes = rayTracingFeatureProcessor->GetProceduralGeometryTypes();
+                    AZStd::unordered_map<Name, unsigned> geometryTypeMap;
+                    geometryTypeMap.reserve(proceduralGeometryTypes.size());
+                    for (auto it = proceduralGeometryTypes.cbegin(); it != proceduralGeometryTypes.cend(); ++it)
+                    {
+                        geometryTypeMap[it->m_name] = proceduralHitGroupIndex++;
+                    }
+
+                    for (const auto& proceduralGeometry : rayTracingFeatureProcessor->GetProceduralGeometries())
+                    {
+                        tlasDescriptorBuild->Instance()
+                            ->InstanceID(instanceIndex)
+                            ->InstanceMask(proceduralGeometry.m_instanceMask)
+                            ->HitGroupIndex(geometryTypeMap[proceduralGeometry.m_typeHandle->m_name])
+                            ->Blas(proceduralGeometry.m_blas)
+                            ->Transform(proceduralGeometry.m_transform)
+                            ->NonUniformScale(proceduralGeometry.m_nonUniformScale);
+                        instanceIndex++;
+                    }
+
+                    // create the TLAS buffers based on the descriptor
+                    RHI::Ptr<RHI::RayTracingTlas>& rayTracingTlas = rayTracingFeatureProcessor->GetTlas();
+                    rayTracingTlas->CreateBuffers(
+                        RHI::RHISystemInterface::Get()->GetRayTracingSupport(), &tlasDescriptor, rayTracingBufferPools);
+                }
+
+                // update and compile the RayTracingSceneSrg and RayTracingMaterialSrg
+                // Note: the timing of this update is very important, it needs to be updated after the TLAS is allocated so it can
+                // be set on the RayTracingSceneSrg for this frame, and the ray tracing mesh data in the RayTracingSceneSrg must
+                // exactly match the TLAS.  Any mismatch in this data may result in a TDR.
+                rayTracingFeatureProcessor->UpdateRayTracingSrgs();
```

Comment:
Dont forget to do a bunch of testing for TDRs based on this comment.

Reply:
We tested this and also investigated the callstack that happens between `FrameBeginInternal` and `SetupFrameGraphDependencies` and did not find anything that should introduce a problem.